 GenAI & Machine Learning Bootcamp 2025 - Full Time 2025 - PSTB Data understanding and preprocessing Data Analysis and Data Types Intro Exercises XP
Exercises XP
Last Updated: June 4th, 2025

What we will learn
Fundamental Concepts and Significance of Data Analysis: Understanding the basics of data analysis and its importance in various contemporary contexts.
Different Data Types in Data Analysis: Learning about qualitative and quantitative data types and how to identify them in datasets.
Basic Data Analysis Techniques: Gaining practical skills in performing basic data analysis tasks using tools like Python, Pandas, Matplotlib, and Seaborn libraries in a Jupyter Notebook or Google Colab environment.


What You Will Create:
An Essay/Report on Data Analysis (Exercise 1): A written piece that defines data analysis, discusses its significance, and explores its applications in today’s world.
Data Type Classification Report (Exercise 2): A detailed analysis of the Iris dataset, categorizing each column as either qualitative or quantitative with accompanying explanations.
A Data Analysis Notebook (Exercise 3): A comprehensive Jupyter notebook that includes calculations of statistical measures (mean, median, mode), data visualizations (histograms, bar charts), and documentation of findings, demonstrating basic data analysis skills.


🌟Exercise 1: Introduction to Data Analysis (Easy)


Objective: Understand the basic overview and significance of data analysis.
Task:
Write a short essay or report on the following topics:

What is data analysis?
Why is data analysis important in modern contexts?
List and describe three areas where data analysis is applied today.
Hint/Tip:

Research current trends in data analysis and real-world examples to provide depth to your essay.


🌟Exercise 2: Dataset Loading and Initial Analysis


Objective: Practice dataset loading from Kaggle and initial analysis.
Task:
for the following dataset : How Much Sleep Do Americans Really Get?, Global Trends in Mental Health Disorder and Credit Card Approvals.

Load the dataset into Jupyter or Google Colab.
Display the first few rows.
Provide a brief dataset description.


🌟Exercise 3: Identifying Data Types


Objective: Learn to identify different data types.
Task:
For the datasets from the previous exercise, categorize each column of it as either quantitative or qualitative and explain your reasoning.



🌟Exercise 4: Exploring Data Types


Objective: Learn about different types of data in data analysis.
Task:
Load the Iris dataset using Kaggle into a Jupyter Notebook or Google Colaboratory Notebook.
Identify and list which columns in your dataset are qualitative and which are quantitative.
Write a brief description of why each column is classified as qualitative or quantitative.
Tools: Jupyter Notebook, Python with Pandas library.



🌟Exercise 5: Basic Data Analysis with Google Colab


Objective: Perform basic data analysis using Google Colab.
Task:
Using the same notebook from the previous exercise, perform basic data analysis tasks:
Calculate the mean, median, and mode of a quantitative column.
Create a simple plot (like a histogram or bar chart) to visualize the data using Matplotlib/Seaborn libraries :
import matplotlib.pyplot as plt
plt.plot(data['Name_column1'], data['Name_column2'])
plt.show()
Document your findings in the notebook.
Tools: Google Colab, Python with Pandas and Matplotlib/Seaborn libraries.



🌟Exercise 6: Basic Observation Skills in Data Analysis


Objective: Develop observation skills for data analysis.
Task:
Load the How Much Sleep Do Americans Really Get? dataset into Jupyter or Google Colab.
Identify columns that could be interesting for a specific type of analysis (e.g., trend analysis, group comparison) and explain your choice.
Tools: Jupyter Notebook, Python with Pandas library.


#Exercise 1: Introduction to Data Analysis (Easy)

#Task: Write a short essay or report on the following topics:

#What is data analysis?
    #In short, data analysis is a process of inspecting, cleansing, transforming, and modeling data with the goal of discovering useful information, #informing conclusions, and supporting decision-making. This process involves a variety of techniques and tools, ranging from basic statistical #analysis to complex algorithms and predictive models.

#Why is data analysis important in modern contexts?
        #Because by supporting in-depth analysis, together with predictive scenarii and complex decision making, it can sustain great value creation.

#List and describe three areas where data analysis is applied today.
        #Health care and new drug making process
        #conception of marketing campaign and communication
        #predictive scenarii and decision making process for Supply Chain, etc.


#Hint/Tip: Research current trends in data analysis and real-world examples to provide depth to your essay.

🌟Exercise 2: Dataset Loading and Initial Analysis


Objective: Practice dataset loading from Kaggle and initial analysis.
Task:
for the following dataset : How Much Sleep Do Americans Really Get?, Global Trends in Mental Health Disorder and Credit Card Approvals.

Load the dataset into Jupyter or Google Colab.
Display the first few rows.
Provide a brief dataset description.

In [1]:
import pandas as pd

In [13]:
#df = pd.read_csv('C:\Users\cibei\Desktop\PSTB GenAI\WEEK3\DAY1\ExerciceXP\Time_Americans_Spend_Sleeping.csv')

df = pd.read_csv('Time_Americans_Spend_Sleeping.csv')



In [3]:
#Load the dataset into Jupyter or Google Colab.
    #Display the first few rows.
df.head()

index  Year  Period  Avg hrs per day sleeping  Standard Error Type of Days  \
0      0  2003  Annual                      8.57           0.018     All days   
1      1  2004  Annual                      8.55           0.026     All days   
2      2  2005  Annual                      8.62           0.023     All days   
3      3  2006  Annual                      8.63           0.024     All days   
4      4  2007  Annual                      8.57           0.024     All days   

           Age Group  Activity   Sex  
0  15 years and over  Sleeping  Both  
1  15 years and over  Sleeping  Both  
2  15 years and over  Sleeping  Both  
3  15 years and over  Sleeping  Both  
4  15 years and over  Sleeping  Both

In [6]:
#Provide a brief dataset description.
df.describe()

index         Year  Avg hrs per day sleeping  Standard Error
count  945.000000   945.000000                945.000000      945.000000
mean   472.000000  2010.000000                  8.806868        0.086958
std    272.942302     4.322782                  0.545001        0.034631
min      0.000000  2003.000000                  7.720000        0.018000
25%    236.000000  2006.000000                  8.360000        0.063000
50%    472.000000  2010.000000                  8.810000        0.084000
75%    708.000000  2014.000000                  9.190000        0.106000
max    944.000000  2017.000000                 10.650000        0.228000

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 945 entries, 0 to 944
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   index                     945 non-null    int64  
 1   Year                      945 non-null    int64  
 2   Period                    945 non-null    object 
 3   Avg hrs per day sleeping  945 non-null    float64
 4   Standard Error            945 non-null    float64
 5   Type of Days              945 non-null    object 
 6   Age Group                 945 non-null    object 
 7   Activity                  945 non-null    object 
 8   Sex                       945 non-null    object 
dtypes: float64(2), int64(2), object(5)
memory usage: 66.6+ KB


In [ ]:
#Exercise 3: Identifying Data Types

#Task: For the datasets from the previous exercise, categorize each column of it as either quantitative or qualitative and explain your reasoning.

In [7]:
#Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype      Comment 
#---  ------                   --------------  -----     --------
# 0   index                     945 non-null    int64     quantitative (int)
# 1   Year                      945 non-null    int64     quantitative (int)
# 2   Period                    945 non-null    object    qualitative (string)
# 3   Avg hrs per day sleeping  945 non-null    float64   quantitative (float)
# 4   Standard Error            945 non-null    float64   quantitative (float)
# 5   Type of Days              945 non-null    object    qualitative (string)
# 6   Age Group                 945 non-null    object    qualitative (string)
# 7   Activity                  945 non-null    object    qualitative (string)
# 8   Sex                       945 non-null    object    qualitative (string)

Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   index                     945 non-null    int64    / quantitatif & numérique
 1   Year                      945 non-null    int64    / quantitatif & numérique
 2   Period                    945 non-null    object   / qualitatif & string
 3   Avg hrs per day sleeping  945 non-null    float64  / quantitatif & numérique
 4   Standard Error            945 non-null    float64  / quantitatif & numérique
 5   Type of Days              945 non-null    object   / qualitatif & string
 6   Age Group                 945 non-null    object   / qualitatif & string
 7   Activity                  945 non-null    object   / qualitatif & string
 8   Sex                       945 non-null    object   / qualitatif & string
dtypes: float64(2), int64(2), object(5)
memory usage: 66.6+ KB

🌟Exercise 4: Exploring Data Types

Objective: Learn about different types of data in data analysis.

Task:
    Load the Iris dataset using Kaggle into a Jupyter Notebook or Google Colaboratory Notebook.
    Identify and list which columns in your dataset are qualitative and which are quantitative.
    Write a brief description of why each column is classified as qualitative or quantitative.
    
    Tools: Jupyter Notebook, Python with Pandas library.

In [9]:
df = pd.read_csv('Iris.csv')


In [10]:
df.head()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa

In [11]:
df.describe()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
count  150.000000     150.000000    150.000000     150.000000    150.000000
mean    75.500000       5.843333      3.054000       3.758667      1.198667
std     43.445368       0.828066      0.433594       1.764420      0.763161
min      1.000000       4.300000      2.000000       1.000000      0.100000
25%     38.250000       5.100000      2.800000       1.600000      0.300000
50%     75.500000       5.800000      3.000000       4.350000      1.300000
75%    112.750000       6.400000      3.300000       5.100000      1.800000
max    150.000000       7.900000      4.400000       6.900000      2.500000

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             150 non-null    int64  
 1   SepalLengthCm  150 non-null    float64
 2   SepalWidthCm   150 non-null    float64
 3   PetalLengthCm  150 non-null    float64
 4   PetalWidthCm   150 non-null    float64
 5   Species        150 non-null    object 
dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             150 non-null    int64           / <class 'pandas.core.frame.DataFrame'>

dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB
 1   SepalLengthCm  150 non-null    float64         / quantitative (number)
 2   SepalWidthCm   150 non-null    float64         / quantitative (number)
 3   PetalLengthCm  150 non-null    float64         / quantitative (number)
 4   PetalWidthCm   150 non-null    float64         / quantitative (number)
 5   Species        150 non-null    object          / qualitative (string)
dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB

🌟Exercise 5: Basic Data Analysis with Google Colab

Objective: Perform basic data analysis using Google Colab.

Task: Using the same notebook from the previous exercise, perform basic data analysis tasks:
    Calculate the mean, median, and mode of a quantitative column.
    Create a simple plot (like a histogram or bar chart) to visualize the data using Matplotlib/Seaborn libraries :
        import matplotlib.pyplot as plt
        plt.plot(data['Name_column1'], data['Name_column2'])
        plt.show()
Document your findings in the notebook.

Tools: Google Colab, Python with Pandas and Matplotlib/Seaborn libraries.

In [ ]:
#Exercise 5: Basic Data Analysis with Google Colab

#Objective: Perform basic data analysis using Google Colab.

#Task: Using the same notebook from the previous exercise, perform basic data analysis tasks:
    #Calculate the mean, median, and mode of a quantitative column.


    





In [ ]:
#Create a simple plot (like a histogram or bar chart) to visualize the data using Matplotlib/Seaborn libraries :
        #import matplotlib.pyplot as plt

import matplotlib.pyplot as plt
plt.plot(data['Name_column1'], data['Name_column2'])
plt.show()



In [ ]:
        #plt.plot(data['Name_column1'], data['Name_column2'])
        #plt.show()

In [ ]:
#Exercise 6: Basic Observation Skills in Data Analysis

#Task:
    #Load the How Much Sleep Do Americans Really Get? dataset into Jupyter or Google Colab.

import pandas as pd


df = pd.read_csv('Time_Americans_Spend_Sleeping.csv')

In [4]:
#Identify columns that could be interesting for a specific type of analysis (e.g., trend analysis, group comparison) and explain your choice. 
##Tools: Jupyter Notebook, Python with Pandas library.

df.head()


index  Year  Period  Avg hrs per day sleeping  Standard Error Type of Days  \
0      0  2003  Annual                      8.57           0.018     All days   
1      1  2004  Annual                      8.55           0.026     All days   
2      2  2005  Annual                      8.62           0.023     All days   
3      3  2006  Annual                      8.63           0.024     All days   
4      4  2007  Annual                      8.57           0.024     All days   

           Age Group  Activity   Sex  
0  15 years and over  Sleeping  Both  
1  15 years and over  Sleeping  Both  
2  15 years and over  Sleeping  Both  
3  15 years and over  Sleeping  Both  
4  15 years and over  Sleeping  Both

In [5]:
df.describe()

index         Year  Avg hrs per day sleeping  Standard Error
count  945.000000   945.000000                945.000000      945.000000
mean   472.000000  2010.000000                  8.806868        0.086958
std    272.942302     4.322782                  0.545001        0.034631
min      0.000000  2003.000000                  7.720000        0.018000
25%    236.000000  2006.000000                  8.360000        0.063000
50%    472.000000  2010.000000                  8.810000        0.084000
75%    708.000000  2014.000000                  9.190000        0.106000
max    944.000000  2017.000000                 10.650000        0.228000

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 945 entries, 0 to 944
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   index                     945 non-null    int64  
 1   Year                      945 non-null    int64  
 2   Period                    945 non-null    object 
 3   Avg hrs per day sleeping  945 non-null    float64
 4   Standard Error            945 non-null    float64
 5   Type of Days              945 non-null    object 
 6   Age Group                 945 non-null    object 
 7   Activity                  945 non-null    object 
 8   Sex                       945 non-null    object 
dtypes: float64(2), int64(2), object(5)
memory usage: 66.6+ KB


#Identify columns that could be interesting for a specific type of analysis (e.g., trend analysis, group comparison) and explain your choice. 
    "Age Group", "Activity", "Sex".
    These are three qualitative criteria that can better support specific types of analysis like trend analysis, group comparison, ...
    The other listed criteria are more or quantitative ones.

#   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  

 6   Age Group                 945 non-null    object 
 7   Activity                  945 non-null    object 
 8   Sex                       945 non-null    object 
dtypes: float64(2), int64(2), object(5)